index.bs

```diff
@@ -1620,20 +1644,22 @@ To <dfn for="WebTransportSendStream">write</dfn> |chunk| to a {{WebTransportSend
 
      Note: The definition of fairness here is [=implementation-defined=].
 
-  1. If the previous step failed, abort the remaining steps.
+  1. If the previous step failed due to a network error, abort the remaining steps.
 
     Note: We don't reject |promise| here because we handle network errors elsewhere, and those steps
     error |stream| and reject the result of this write operation.
 
-  1. [=Queue a network task=] with |transport| to run these steps:
+  1. Otherwise, [=queue a network task=] with |transport|
+     to run these steps:
     1. Set |stream|.{{[[PendingOperation]]}} to null.
+    1. If |stream|.{{WebTransportSendStream/[[AtomicWriteRequests]]}} [=list/contains=] |inFlightWriteRequest|, [=list/remove=] |inFlightWriteRequest|.
     1. [=Resolve=] |promise| with undefined.
 1. Return |promise|.
 
 Note: The [=fulfilled|fulfillment=] of the promise returned from this algorithm (or,
-{{WritableStreamDefaultWriter/write|WritableStreamDefaultWriter.write}}) does **NOT** necessarily mean that the chunk is acked by
+{{WritableStreamDefaultWriter/write(chunk)}}) does **NOT** necessarily mean that the chunk is acked by
 the server [[!QUIC]]. It may just mean that the chunk is appended to the buffer. To make sure that
-the chunk arrives at the server, use an application-level protocol.
+the chunk arrives at the server, the server needs to send some form of acknowledgment message.
```

Comment:
```suggestion
the chunk arrives at the server, the server needs to send an application-level acknowledgment message.
```
I think we should keep "application-level" to make it clear that a transport-level ack is not sufficient.